### PR TITLE
ENH: Revise externalproject example, extend ElastixConfig.cmake file

### PR DESCRIPTION
**Pull request by N-Dekker**
Added `ELASTIX_ITK_DIR` CMake variables to ElastixConfig.cmake, which appears necessary.

Replaced the "TimerTest" by an example that does a simple 2-D registration.

Added to GitHubActions and Azure CI -- with help from Viktor van der Valk (@ViktorvdValk)

### CODE REVIEW

**Comment by N-Dekker** [comment]
_Expected output:_

Image registation example: estimating the translation between two images of 5x6 pixels

Linked to elastix library version 5.0.1
Web site: https://elastix.lumc.nl
Source code repository: https://github.com/SuperElastix/elastix
Discussion forum: https://groups.google.com/g/elastix-imageregistration

Fixed image:

        0 0 0 0 0
        0 0 0 0 0
        0 0 0 0 0
        0 0 3 0 0
        0 4 5 0 0
        0 0 0 0 0

Moving image:

        0 0 0 0 0
        0 0 0 3 0
        0 0 4 5 0
        0 0 0 0 0
        0 0 0 0 0
        0 0 0 0 0

Estimated translation:

        1 -2

**Comment by N-Dekker** [comment]
🎉  Finally made Windows, macOS _and_ Ubuntu at GitHubActions happy with this very, very simple example project! Please check, especially the CMake file of the external (example) project: https://github.com/SuperElastix/elastix/blob/7916198892908c60cada0b1897ffc27e5229938f/dox/externalproject/CMakeLists.txt 

As you see, it took a lot of trial-and-error to get it all building.

**Comment by mstaring** [comment]
great work!

File: dox/externalproject/CMakeLists.txt [hunk 1]
```diff
@@ -1,29 +1,27 @@
...
-SET( ELASTIX_USE_FILE ${ELASTIX_BINARY_DIR}/UseElastix.cmake )
-IF( EXISTS ${ELASTIX_USE_FILE} )
- MESSAGE( STATUS "Including Elastix settings." )
- INCLUDE( ${ELASTIX_USE_FILE} )
-ENDIF()
+# Elastix might use OpenMP.
+find_package( OpenMP QUIET )
+if( OPENMP_FOUND )
+  set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}" )
+endif()
```
Comment (N-Dekker) [inline dox/externalproject/CMakeLists.txt:12]: This was one of the hardest parts for me to figure out: _if_ elastix might use OpenMP, the openmp flag must be added to the compiler flags of the external project! 

On the CI, OpenMP is "found" (by CMake find_package) on Windows and Ubuntu, but not on  macOS.

File: dox/externalproject/CMakeLists.txt [hunk 2]
```diff
@@ -1,29 +1,27 @@
...
-# the one found in the <elastix-dir>/Testing dir.
-ADD_EXECUTABLE( elxtimertest itkTimerTest.cxx )
+# Use the version of ITK from Elastix.
+set( ITK_DIR ${ELASTIX_ITK_DIR} )
+find_package( ITK REQUIRED )
 
-# Link to some libraries
-TARGET_LINK_LIBRARIES( elxtimertest
-  ITKCommon ITKBasicFilters elxCommon )
+include(${ELASTIX_CONFIG_TARGETS_FILE})
```
Comment (N-Dekker) [inline dox/externalproject/CMakeLists.txt:18]: Matt (@thewtex) suggested including the Elastix Target file ("ElastixTargets.cmake"). This appeared essential in order to get the file names of the dependencies right (including possible, OS-specific "lib" prefix, and elastix specifix version number suffix.)

**Pull request merged**